src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php

```diff
@@ -40,6 +42,23 @@ public function testInstantiateObjectDenormalizer()
         $this->assertInstanceOf(__NAMESPACE__.'\Dummy', $normalizer->instantiateObject($data, $class, $context, new \ReflectionClass($class), array()));
     }
 
+    public function testDenormalizeWithDiscriminatorAndPropertyInfo()
+    {
+        $propertyTypeExtractor = $this->createMock(PropertyTypeExtractorInterface::class);
+        $propertyTypeExtractor->expects($this->exactly(2))
+            ->method('getTypes')
+            ->withConsecutive(
+                array(MappedDummyChild::class, 'foo'),
+                array(MappedDummyChild::class, 'bar')
+            );
+
+        $normalizer = new AbstractObjectNormalizerWithMetadata($propertyTypeExtractor);
+
+        $data = array('foo' => 'dummy', 'bar' => 1);
+        $class = MappedDummy::class;
+        $normalizer->denormalize($data, $class);
```

Comment:
Can you also assert on the result of the `denormalize` method? At the end, that's what you want to test, isn't it?

Reply:
Because at the end of the day... your example without your change return the following object, which is what you expect 🤔 

```php
class Symfony\Component\Serializer\Tests\Normalizer\MappedDummyChild#1403 (3) {
  public $bar =>
  int(1)
  public $foo =>
  string(5) "dummy"
  public $baz =>
  NULL
}
```